File: app/src/main/java/com/tao/chinachuclient/ProgramListClickListener.java
package com.tao.chinachuclient;

import Chinachu4j.Program;
import Chinachu4j.Recorded;
import Chinachu4j.Reserve;
import android.content.Context;
import android.content.Intent;
import android.view.View;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;

public class ProgramListClickListener implements OnItemClickListener{

	private Context context;
	private int type;

	public ProgramListClickListener(Context context, int type){
		this.context = context;
		this.type = type;
	}

	@Override
	public void onItemClick(AdapterView<?> parent, View view, int position, long id){
		Intent i = new Intent(context, ProgramDetail.class);
		if(type == Type.RESERVES)
			i.putExtra("reserve", (Reserve)parent.getItemAtPosition(position));
		else if(type == Type.RECORDED)
			i.putExtra("recorded", (Recorded)parent.getItemAtPosition(position));
		else
			i.putExtra("program", (Program)parent.getItemAtPosition(position));
		i.putExtra("type", type);
		context.startActivity(i);
	}
}

File: app/src/main/java/com/tao/chinachuclient/ProgramListClickListener.kt
package com.tao.chinachuclient

import android.content.Context
import android.content.Intent
import android.view.View
import android.widget.AdapterView
import com.tao.chinachuclient.ui.programdetail.ProgramDetailActivity
import sugtao4423.library.chinachu4j.Program
import sugtao4423.library.chinachu4j.Recorded
import sugtao4423.library.chinachu4j.Reserve

class ProgramListClickListener(private val context: Context, private val type: Int) :
    AdapterView.OnItemClickListener {

    override fun onItemClick(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
        if (parent == null) {
            return
        }

        val i = Intent(context, ProgramDetailActivity::class.java)
        when (type) {
            Type.RESERVES -> {
                i.putExtra("reserve", (parent.getItemAtPosition(position) as Reserve))
            }
            Type.RECORDED -> {
                i.putExtra("recorded", (parent.getItemAtPosition(position) as Recorded))
            }
            else -> {
                i.putExtra("program", (parent.getItemAtPosition(position) as Program))
            }
        }
        i.putExtra("type", type)
        context.startActivity(i)
    }

}
